src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp

```diff
@@ -5309,7 +5310,9 @@ CTranslatorExprToDXL::PdxlnScCmpPartKey
 			pexprPartKey->Release();
 		}
 
-		CTranslatorExprToDXLUtils::ExtractCastMdids(pexprNewPartKey->Pop(), &pmdidTypeCastExpr, &mdid_cast_func);
+		//Extract cast/func details
+		CTranslatorExprToDXLUtils::ExtractCastFuncMdids(pexprNewPartKey->Pop(), &pmdidTypeCastExpr, &mdid_cast_func);
+		BOOL is_allowed_lossy_cast = (COperator::EopScalarFunc == pexprPartKey->Pop()->Eopid());
```

Comment:
can you add a comment for clarity or an assert such as  
```suggestion
		BOOL is_allowed_lossy_cast = (COperator::EopScalarFunc == pexprPartKey->Pop()->Eopid());
		GPOS_ASSERT_IMP(COperator::EopScalarFunc == pexprPartKey->Pop()->Eopid(), FAllowedFuncScId(pexprPartKey)); 
```

OR 
``` suggestion 
		BOOL is_allowed_lossy_cast = false;
		if (COperator::EopScalarFunc == pexprPartKey->Pop()->Eopid())
		{
				GPOS_ASSERT(FAllowedFuncScId(pexprPartKey));
				is_allowed_lossy_cast = true;
		};

Reply:
We call `ExtractComponents` from `CPredicateUtils`  earlier in this caller where this check is already done